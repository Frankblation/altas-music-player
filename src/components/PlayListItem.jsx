import React from "react"

export default function PlayListItem({ song }) {
  return (
    <li className="flex justify-between items-center py-2 border-b last:border-b-0">
      <div>
        <h3 className="font-semibold">{song.title}</h3>
        <p className="text-sm text-gray-600">{song.artist}</p>
      </div>
      <span className="text-sm text-gray-600">{song.duration}</span>
    </li>
  )
}

