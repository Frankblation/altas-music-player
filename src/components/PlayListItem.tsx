// PlayListItem.tsx

interface PlayListItemProps {
  title: string;
  artist: string;
  length: string;
  isSelected: boolean;
  onClick?: () => void;
}

const PlayListItem = ({ title, artist, length, isSelected, onClick }: PlayListItemProps) => {
  return (
    <div
      className={`flex rounded-md shadow-sm justify-between items-center p-3 dark:border-m-cyan-900 ${isSelected ? "bg-blue-100" : "hover:bg-gray-100"}`}
      onClick={onClick}
    >
      <div>
        <h3 className="font-medium text-gray-800">{title}</h3>
        <p className="text-sm text-gray-600">{artist}</p>
      </div>
      <span className="text-sm text-gray-500">{length}</span>
    </div>
  )
}

export default PlayListItem