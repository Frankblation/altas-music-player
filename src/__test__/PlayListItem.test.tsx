import { render } from "@testing-library/react";
import { expect, test } from "vitest";
import PlayListItem from "../components/PlayListItem";

test("PlayListItem renders with title and artist", () => {
  const { container } = render(<PlayListItem title="Song A" artist="Artist A" />);
  expect(container).toMatchSnapshot();
});

test("PlayListItem renders without artist", () => {
  const { container } = render(<PlayListItem title="Song B" artist="" />);
  expect(container).toMatchSnapshot();
});
