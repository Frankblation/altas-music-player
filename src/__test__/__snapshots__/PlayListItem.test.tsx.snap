// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PlayListItem renders with title and artist 1`] = `
<div>
  <div
    class="flex rounded-md shadow-sm justify-between items-center p-3 dark:border-m-cyan-900 hover:bg-gray-100"
  >
    <div>
      <h3
        class="font-medium text-gray-800"
      >
        Song A
      </h3>
      <p
        class="text-sm text-gray-600"
      >
        Artist A
      </p>
    </div>
    <span
      class="text-sm text-gray-500"
    />
  </div>
</div>
`;

exports[`PlayListItem renders without artist 1`] = `
<div>
  <div
    class="flex rounded-md shadow-sm justify-between items-center p-3 dark:border-m-cyan-900 hover:bg-gray-100"
  >
    <div>
      <h3
        class="font-medium text-gray-800"
      >
        Song B
      </h3>
      <p
        class="text-sm text-gray-600"
      />
    </div>
    <span
      class="text-sm text-gray-500"
    />
  </div>
</div>
`;
